api/app_analytics/influxdb_wrapper.py

```diff
@@ -149,14 +154,20 @@ def get_event_list_for_organisation(organisation_id: int, date_range: str = "30d
         filters=f'|> filter(fn:(r) => r._measurement == "api_call") \
                   |> filter(fn: (r) => r["organisation_id"] == "{organisation_id}")',
         extra="|> aggregateWindow(every: 24h, fn: sum)",
-        date_range=date_range,
+        date_start=date_start,
+        date_stop=date_stop,
     )
     dataset = defaultdict(list)
     labels = []
     for result in results:
         for record in result.records:
             dataset[record["resource"]].append(record["_value"])
-            required_records = int(date_range[:-1]) + 1
+            if date_stop == "now()":
+                required_records = abs(int(date_start[:-1])) + 1
+            else:
+                required_records = (
+                    abs(int(date_start[:-1])) - abs(int(date_stop[:-1])) + 1
+                )
             if len(labels) != required_records:
```

Comment:
I'm not sure what cases this might happen?

Reply:
It's existing code and I'm not sure when `labels` wouldn't align with the `required_records`. My best guess is that it's trying to add time strings for records that are previously unseen.